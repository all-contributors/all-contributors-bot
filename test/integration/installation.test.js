const Stream = require("stream");

const { Probot, ProbotOctokit } = require("probot");
const pino = require("pino");

const app = require("../../app");

// fixtures
const installationCreated = require("../fixtures/installation.created.json");

let output = [];
const streamLogsToOutput = new Stream.Writable({ objectMode: true });
streamLogsToOutput._write = (object, encoding, done) => {
  const { pid, time, hostname, ...data } = JSON.parse(object);
  output.push(data);
  done();
};

describe("installation event", () => {
  let probot;

  beforeEach(async () => {
    output = [];
    probot = new Probot({
      githubToken: "test",
      Octokit: ProbotOctokit.defaults({
        log: pino(streamLogsToOutput),
        // Disable throttling & retrying requests for easier testing
        retry: { enabled: false },
        throttle: { enabled: false },
      }),
      log: pino(streamLogsToOutput),
    });

    await probot.load(app);
  });

  test("installation.created", async () => {
    await probot.receive({
      name: "installation",
      payload: installationCreated,
    });

    expect(output).toMatchSnapshot();
  });
});
