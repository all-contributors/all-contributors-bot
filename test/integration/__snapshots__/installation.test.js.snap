// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installation event installation.created 1`] = `
Array [
  Object {
    "account": 2001,
    "accountLogin": "octocat",
    "accountType": "user",
    "action": "created",
    "event": "installation",
    "installation": 1,
    "level": 30,
    "msg": "octocat: installation created",
    "name": "event",
    "repositoriesChange": 3,
    "selection": "all",
  },
]
`;

exports[`installation event installation.deleted 1`] = `
Array [
  Object {
    "account": 2001,
    "accountLogin": "octocat",
    "accountType": "user",
    "action": "deleted",
    "event": "installation",
    "installation": 1,
    "level": 30,
    "msg": "octocat: installation deleted",
    "name": "event",
    "repositoriesChange": -3,
    "selection": "all",
  },
]
`;

exports[`installation event installation.suspend 1`] = `
Array [
  Object {
    "account": 2001,
    "accountLogin": "octocat",
    "accountType": "user",
    "action": "suspend",
    "event": "installation",
    "installation": 1,
    "level": 30,
    "msg": "octocat: installation suspend",
    "name": "event",
    "repositoriesChange": 0,
    "selection": "all",
  },
]
`;

exports[`installation event installation_repositories.added 1`] = `
Array [
  Object {
    "account": 2001,
    "accountLogin": "octocat",
    "accountType": "user",
    "action": "added",
    "event": "installation_repositories",
    "installation": 1,
    "level": 30,
    "msg": "octocat: installation_repositories added",
    "name": "event",
    "repositoriesChange": 1,
    "selection": "all",
  },
]
`;
